### PR TITLE
feat: Add loading-bar component

### PR DESCRIPTION
**Pull request by gethinwebster**
### Description

Add the (generative AI) loading bar component

Related links, issue #, if available: n/a

### How has this been tested?

Unit tests & dev pages

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/loading-bar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+// SPDX-License-Identifier: Apache-2.0
+import useBaseComponent from "../internal/base-component/use-base-component";
+import { applyDisplayName } from "../internal/utils/apply-display-name";
+import type { LoadingBarProps } from "./interfaces";
+import { InternalLoadingBar } from "./internal";
+
+export type { LoadingBarProps };
+
+export default function LoadingBar(props: LoadingBarProps) {
+  const baseComponentProps = useBaseComponent("LoadingBar");
```
Comment (abdhalees) [inline src/loading-bar/index.tsx:11]: do we want to collect property metrics for this component? 

```suggestion
  const baseComponentProps = useBaseComponent("LoadingBar", {variant: props.variant});
```

File: src/loading-bar/internal.tsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import styles from "./styles.css.js";
+
+export function InternalLoadingBar({
+  variant,
+  __internalRootRef,
+  ...rest
+}: LoadingBarProps & InternalBaseComponentProps) {
+  return (
+    <div
+      ref={__internalRootRef}
```
Comment (abdhalees) [inline src/loading-bar/internal.tsx:17]: what's this used for?

File: src/loading-bar/styles.scss [hunk 3]
```diff
@@ -0,0 +1,59 @@
+@use "../../node_modules/@cloudscape-design/design-tokens/index.scss" as awsui;
```
Comment (abdhalees) [inline src/loading-bar/styles.scss:1]:
```suggestion
@use "~@cloudscape-design/design-tokens/index.scss" as awsui;```

File: src/loading-bar/internal.tsx (hunk 2 above)
Comment (gethinwebster) [reply src/loading-bar/internal.tsx:17]: `__internalRootRef` is required for metrics

File: src/loading-bar/styles.scss (hunk 3 above)
Comment (gethinwebster) [reply src/loading-bar/styles.scss:1]: we don't currently have the right build tool config for this (see also e.g. https://github.com/cloudscape-design/board-components/blob/main/src/board/styles.scss#L1), will create a follow up to try to tidy this up across all sub-repositories

**Review by abdhalees — APPROVED** [review]

**Pull request merged**